src/transformStateWithClones.js

```
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        stateCopy = {};
+        break;
+      case 'addProperties':
+        stateCopy = { ...stateCopy, ...action.extraData };
+        break;
+      case 'removeProperties':
+        stateCopy = { ...stateCopy };
+
+        if (Array.isArray(action.keysToRemove)) {
+          for (const key of action.keysToRemove) {
+            delete stateCopy[key];
+          }
+        }
+        break;
+      default:
+        stateCopy = { ...stateCopy };
+        break;
```

This violates checklist item #3: "switch/case should always have default case for error handling." 

Right now the `default` case simply clones the current state and continues (lines 30–32). The checklist expects the default case to perform error handling for unknown action types. Consider failing fast with a clear error so callers are informed of invalid actions. For example:

```javascript
default:
  throw new Error(`Unknown action type: ${action.type}`);
```

Place this inside the `switch` (replacing the current default) so unknown action types are explicitly handled rather than silently ignored.